src/main/scala/scodec/stream/decode/StreamDecoder.scala

```diff
@@ -37,11 +35,11 @@ trait StreamDecoder[+A] { self =>
     decode[IO](bits).compile.toVector.unsafeRunSync
 
   /**
-   * Decoding a stream of `A` values from the given `BitVector`.
+   * Decode a stream of `A` values from the given `BitVector`.
    * This function does not retain a reference to `bits`, allowing
    * it to be be garbage collected as the returned stream is traversed.
    */
-  final def decode[F[_]](bits: => BitVector)(implicit F: Effect[F]): Stream[F,A] = Stream.suspend {
+  final def decode[F[_]](bits: => BitVector)(implicit F: Sync[F]): Stream[F,A] = Stream.suspend {
     @volatile var cur = bits // am pretty sure this doesn't need to be volatile, but just being safe
     decoder.translate(new (Cursor ~> F) {
       def apply[X](c: Cursor[X]): F[X] = F.suspend {
```

Comment:
Why is `F.suspend` needed here btw? stack safety?

Reply:
Actually, I don't know :) I tried removing it and tests hung, but then later I realized that was just due to some generators that generated very large lists. I'll try again.

Reply:
OK yeah it works fine so I lifted the `Sync` constraint to `ApplicativeError`

Reply:
Nice, I've seen the `decodePipeByte` too :)